src/UI/Component/Input/Field/Date.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/* Copyright (c) 2018 Nils Haagen <nils.haagen@concepts-and-training.de> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\UI\Component\Input\Field;
+
+/**
+ * This describes the date-field.
+ */
+interface Date extends Input {
+
+	/**
+	 * Get an input like this using the given format.
+	 */
+	public function withFormat(string $format) : Date;
```

Comment:
This method is of crucial importance, it defines the kind of data that is accepted here (time, dates, both). This should be made more explicit here. Add some more explanation. Maybe even some little dummy example.